### ISSUE TITLE
Can we have more styling options? 

### ISSUE THREAD

**Issue by rafcamlet** [ziontee113/neo-minimap#2]
Hi! I really like the idea of this plugin and I think I will use it often. :) 

Could you add (in the future, not necessary now) more configuration options for styling the minimap window? You know, some basic staff like highlight groups, borders etc. ? Basically, things that go into nvim_win_set_option and nvim_open_win. 

Thanks in advance!

**Comment by ziontee113** [comment]
Hi, thank you for the suggestion, I'll look into it tomorrow.

**Comment by ziontee113** [comment]
Hi @rafcamlet , I've implemented it in 0886799fe9fd5e881a0c35435d9b2a5fba83c99e. Please tell me your experience using it. Thank you.

**Comment by rafcamlet** [comment]
Thanks! Works for me. 👍 
But you could consider preparing more user-friendly configuration settings. I wouldn't assume that all neovim users are familiar with its api.

**Comment by ziontee113** [comment]
Hi @rafcamlet . Which default settings do you think we should implement?

**Comment by rafcamlet** [comment]
Let me think. Probably float window content and border highlight (by the way `Normal:Normal,FloatBorder:Normal` is quite popular default option) - mostly because users may not be familiar with 'winhighlight' syntax, the border style itself (single, double, etc.) , zindex maybe - but that more informational, so they will know what to adjust if there is a problem. 
If you are planing to add some layout features (like side panels), it would be also good place for this in the future. 

Btw. It's not related to window options, but since we are talking about settings, toggleable feature for removing leading spaces would be nice. :) Something like `str = str:gsub('^ +', '')`

**Comment by ziontee113** [comment]
Hmm... We're currently using `winhl = "Normal:"`, is it good enough? :thinking: 

> Toggle feature for removing leading spaces

I thought about that feature, however I didn't implement it due to the "loss of indentation structure", I mean it'll be harder for the user to recognize the structure and context of the line / map.

**Comment by rafcamlet** [comment]
> Hmm... We're currently using winhl = "Normal:", is it good enough? 🤔

It is enough, just a thing that I saw in some plugins. And the first thing I've changed. :) 

> (...) , however I didn't implement it due to the "loss of indentation structure"

That's why I suggested it to be toggleable by settings. There are cases where I don't care about indentation structure. For example the map of all named functions in the file, without scopes or anything. So we could have different mappings for different things also for those where indentation only make things less readable.

Edit: I realized this might not be clear. I am going to use multiple different minimaps for one file type.
